aptos-node/src/lib.rs

```diff
@@ -384,6 +397,46 @@ fn setup_state_sync_storage_service(
     storage_service_runtime
 }
 
+async fn periodic_telemetry_dump(node_config: NodeConfig, db: DbReaderWriter) {
+    use futures::stream::StreamExt;
+    let mut dump_interval =
+        IntervalStream::new(tokio::time::interval(std::time::Duration::from_secs(30))).fuse();
+
+    info!("periodic_telemetry_dump task started");
+
+    loop {
+        futures::select! {
+            _ = dump_interval.select_next_some() => {
+
+                // Build the params from internal prometheus metrics
+                let mut metrics_params: HashMap<String, String> = HashMap::new();
+
+                // Measurement Protocol params must be underscore or alphanumeric
+                // Prometheus metrics use {} to represent dimensions, so rename it
+                let met = get_public_metrics();
+                let re = Regex::new(r"[\{\}]").unwrap();
+                for (k, v) in &met {
+                    metrics_params.insert(re.replace_all(k, "_").to_string(), v.to_string());
+                }
+                let met = get_public_json_metrics();
+                for (k, v) in &met {
+                    metrics_params.insert(k.to_string(), v.to_string());
+                }
+
+                // get some data we do not currently have metrics for
+                let chain_id = fetch_chain_id(&db);
+                let peer_id = match node_config.peer_id() {
+                    Some(p) => p.to_string(),
+                    None => String::new()
+                };
+                metrics_params.insert("CHAIN_ID".to_string(), chain_id.to_string());
```

Comment:
I'd start creating a contstants.rs in the telemetry crate. That way we can reference these labels from a single place 😄

Reply:
yup just did that!